ts/model/ElasticSearchQuery.ts

```diff
@@ -0,0 +1,156 @@
+    public endDate: Date;
+    public pageSize: number;
+    public page: number;
+
+    constructor(page = 0, pageSize = 250, tenantId?) {
+        this.page = page;
+        this.pageSize = pageSize;
+        this.tenantId = tenantId;
+    }
+
+    public withDateRange(start: Date, end: Date): ElasticSearchQuery {
+        this.startDate = start;
+        this.endDate = end;
+        return this;
+    }
+}
+
+export class ElasticsearchQueryField<K, V> {
+    public key: K;
+    public value: V;
+    public contains: boolean = true;
+    public literal: boolean = true;
+    public queryType: string;
+
+    constructor(key, value, literal = false, contains = true) {
+        this.key = key;
+        this.value = value;
+        this.contains = contains;
+        this.literal = literal;
+    }
+}
+
+export class ElasticSearchQueryItem<T> {
+    public literal: boolean = true;
+    public contains: boolean = true;
+    public queryType: string;
+    public item: T;
+
+    constructor(item, literal = false, contains = true) {
+        this.item = item;
+        this.literal = literal;
+        this.contains = contains;
+    }
+}
+
+export class ElasticSearchQueryList<T> {
+    public static withItems<T>(items: T[], and = false, literal = false, contains = true): ElasticSearchQueryList<T> {
+        const queryItems = items.map((item) => new ElasticSearchQueryItem<T>(item, literal, contains));
+
+        return new ElasticSearchQueryList<T>(queryItems, and);
+    }
+
+    public static withItem<T>(item: T, and = false, literal = false, contains = true): ElasticSearchQueryList<T> {
+        const queryItems = [new ElasticSearchQueryItem<T>(item, literal, contains)];
+
+        return new ElasticSearchQueryList<T>(queryItems, and);
+    }
+
+    public and: boolean = true;
+    public items: Array<ElasticSearchQueryItem<T>>;
+    public nestedItems: Array<ElasticSearchQueryList<T>>;
+
+    constructor(items: Array<ElasticSearchQueryItem<T>>, and = true, nestedItems?) {
+        this.items = items;
+        this.and = and;
+        this.nestedItems = nestedItems;
+    }
+}
+
+export interface IHash<V> {
+    [key: string]: V;
+}
+
+export class ElasticSearchQueryMap<K, V> {
+    public static withItems<K, V>(items: Array<IHash<V>>, keyField: string, valueField: string, and = false, literal = true, contains = true): ElasticSearchQueryMap<K, V> {
+        const queryItems = items.map((entry) => new ElasticsearchQueryField<K, V>(entry[keyField], entry[valueField], literal, contains));
+
+        return new ElasticSearchQueryMap<K, V>(queryItems, and);
+    }
+
+    public and: boolean = true;
+    public items: Array<ElasticsearchQueryField<K, V>> = [];
+    public nestedItems: Array<ElasticSearchQueryMap<K, V>> = [];
+
+    constructor(items: Array<ElasticsearchQueryField<K, V>>, and = true, nestedItems?) {
+        this.items = items;
+        this.and = and;
+        this.nestedItems = nestedItems;
+    }
+}
+
+export class ElasticSearchSortField {
```

Comment:
I only see three of these fields used in this PR, are the other fields necessary?